DisCatSharp.ApplicationCommands/Workers/RegistrationWorker.cs

```diff
@@ -0,0 +1,382 @@
+                }
+            }
+            else
+            {
+                foreach (var cmd in discord)
+                {
+                    if (!updateList.Any(ul => ul.Name == cmd.Name))
+                        InvalidCommandIds.Add(cmd.Id);
+                }
+            }
+
+            return InvalidCommandIds;
+        }
+
+        /// <summary>
+        /// Builds a list of guild commands to be created on discords backend.
+        /// </summary>
+        /// <param name="guildId">The guild id these commands belong to.</param>
+        /// <param name="updateList">The command list.</param>
+        /// <returns></returns>
+        private static List<DiscordApplicationCommand> BuildGuildCreateList(ulong guildId, List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._guildDiscordCommands == null || !ApplicationCommandsExtension._guildDiscordCommands.Any()
+                || updateList == null || ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value == null
+            )
+                return updateList;
+
+            var discord = ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value;
+
+            List<DiscordApplicationCommand> NewCommands = new();
+
+            foreach (var cmd in updateList)
+            {
+                if (!discord.Any(d => d.Name == cmd.Name))
+                {
+                    NewCommands.Add(cmd);
+                }
+            }
+
+            return NewCommands;
+        }
+
+        /// <summary>
+        /// Builds a list of guild commands to be overwritten on discords backend.
+        /// </summary>
+        /// <param name="guildId">The guild id these commands belong to.</param>
+        /// <param name="updateList">The command list.</param>
+        /// <returns>A dictionary of command id and command.</returns>
+        private static Dictionary<ulong, DiscordApplicationCommand> BuildGuildOverwriteList(ulong guildId, List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._guildDiscordCommands == null || !ApplicationCommandsExtension._guildDiscordCommands.Any()
+                || !ApplicationCommandsExtension._guildDiscordCommands.Any(l => l.Key == guildId) || updateList == null
+                || ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value == null
+            )
+                return null;
+
+            var discord = ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value;
+
+            Dictionary<ulong, DiscordApplicationCommand> UpdateCommands = new();
+
+            foreach (var cmd in updateList)
+            {
+                if (discord.Any(d => d.Name == cmd.Name))
+                {
+                    UpdateCommands.Add(discord.Where(d => d.Name == cmd.Name).First().Id, cmd);
```

Comment:
could make this a bit more optimal. You're iterating over the collection once to identity IF something has said name... then again to find that item... essentially finding it twice.